lib: dk_buttons_and_leds: Add no interrupts mode for buttons (#14285)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * Added the :c:func:`date_time_now_local` function to the API.

* :ref:`dk_buttons_and_leds_readme` library:

>   * Added an experimental no interrupts mode for button handling.

Reviewer: Somehow, the name of this mode "no interrupts" does not stand out from the text. Still, I don't really know whether we can highlight in any way. @annwoj, any suggestions?

Reviewer 2: That was on my mind as well earlier, but using any typographical conventions here doesn't really make much sense. We could capitalize it but it won't align with other similar occurrences in the ncs documentation, so I don't really have any good solution for this :(. Figured that the best way not to confuse anyone is leaving is as is. 
Now I just thought of "no-interrupts mode" to highlight this part, and although there are hyphenated usages of this out there I'm not sure how justified it is in terms of punctuation.

>     You can enable it with the :kconfig:option:`CONFIG_DK_LIBRARY_BUTTON_NO_ISR` Kconfig option.
>     It is a workaround that can be used to avoid using the GPIO interrupts, however, it increases the power consumption.

Reviewer: ```suggestion
    To enable this mode, use the :kconfig:option:`CONFIG_DK_LIBRARY_BUTTON_NO_ISR` Kconfig option.
    You can use this mode as a workaround to avoid using the GPIO interrupts.
    However, using it increases the power consumption.
```

Reviewer 2: Can we change from "using" to e.g., 'However, doing so increases...'  (?) as this edit causes the word to repeat 3 times within 2 sentences

Reviewer: You are right, I noticed this. Maybe just drop the word "using" from the last sentence.

Reviewer 2: Agreed! So let's go back to the previous version but keep it split as per suggestion into 2 sentences.

  * Added the :c:func:`date_time_now_local` function to the API.

> * :ref:`dk_buttons_and_leds_readme` library:

Reviewer 3: Can we have the changelog entry into a separate commit? (I know we usually commit such small changes in one commit. However, in this context, we have a chance for a clear revert of this feature once the PDK v0.2.x support is dropped).

>   * Added experimental no interrupts mode for button handling, which can be enabled with the :kconfig:option:`CONFIG_DK_LIBRARY_BUTTON_NO_ISR` Kconfig option.

Reviewer 4: Maybe we could mention here that it increases power consumption and was added specifically for nRF54L PDK v0.2.1 (and rather should not be used for other boards)? @kapi-no  what do you think?

Author: Added only info that this is workaround for nrf54l pdk rev 0.2.x issues, more details can be checked in the kconfig help

  * Added the :c:func:`date_time_now_local` function to the API.

* :ref:`dk_buttons_and_leds_readme` library:

>   * Added experimental no interrupts mode for button handling, which can be enabled with the :kconfig:option:`CONFIG_DK_LIBRARY_BUTTON_NO_ISR` Kconfig option.
>     It is a workaround for the nRF54L15 PDK revision 0.2.X, where Button 3 and 4 are connected to the GPIO port which does not support interrupts.

Reviewer 2: ```suggestion
  * Added an experimental no interrupts mode for button handling. 
    You can enable it with the :kconfig:option:`CONFIG_DK_LIBRARY_BUTTON_NO_ISR` Kconfig option.
    It is a workaround that can be used to avoid using the GPIO interrupts, however, it increases the consumption.
```
